### DISCUSSION TITLE
'Message' object has no attribute 'note'

### DISCUSSION THREAD

**Discussion by Anon1230987654** [mido/mido#385] [Q&A] [answered]
Hello, I am currently trying to get the note of the midi files, but unfortunately I keep getting this error:
'Message' object has no attribute 'note'
Here's my code:
for x in mid.tracks[2]:
    if(not x.is_meta):
        print (x.type, x.time, x.note)

Thanks for taking the time to read my question :)

**Comment by r1yk** [comment, marked as answer]
@Anon1230987654 You're probably getting that error on a type of MIDI message that doesn't have any value for 'note'. In fact, it looks the only message types that _do_ have a value for 'note' are `note_on`, `note_off` and `polytouch`. Those available attributes appear to be defined in the code here:
https://github.com/mido/mido/blob/main/mido/messages/specs.py#L35

Here's a modification of your code that may be helpful for inspecting whatever's going on with your MIDI file. It checks that the message does in fact have a note before printing it out. Hope this helps!
```python
for x in mid.tracks[2]:
    if not x.is_meta:
        if hasattr(x, "note"):
            print(x.type, x.time, x.note)
        else:
            print("Message type without note:", x.type)
```

**Reply by rolfiepolfie** [reply]
Do you have any thoughts about the speed of that hasattr() comparing to the speed of a simple test like: msg.type == 'note_on' or msg.type == 'note_off' for preliminary check before accessing the note attribute?

**Reply by rdoursenaud** [reply]
Some food for thought: https://stackoverflow.com/questions/903130/hasattr-vs-try-except-block-to-deal-with-non-existent-attributes
